torch/utils/_pytree.py

```diff
@@ -120,6 +122,59 @@ class _SerializeNodeDef(NamedTuple):
 SERIALIZED_TYPE_TO_PYTHON_TYPE: Dict[str, Type[Any]] = {}
 
 
+def register_pytree_node(
```

Comment:
@XuehaiPan, I'm reading the code again. `register_pytree_node` should error out on duplicate registrations. However, `_register_pytree_node` should not error out on duplicate registrations, to maintain BC. This doesn't appear to be the case, and this is what is tripping the internal tests.